### DISCUSSION TITLE
hurl $directory

### DISCUSSION THREAD

**Discussion by cassepipe** [Orange-OpenSource/hurl#1443] [Ideas] [answered]
I am putting now puttting my hurl files into a directory and thought it would be a great feature to pass `hurl` a directory so that it would run all the `.hurl` files that are in it.

What do you think ?

Thanks for making your tool open source. 
The command line people is grateful.

**Comment by jcamiel** [comment]
Hi, thanks for your feedbacks!

if you're using bash/zsh etc... you can simply launch Hurl with the following command:

```shell
$ hurl --test mydir/*.hurl
```

There is also the [`--glob` option](https://hurl.dev/docs/manual.html#glob) which can be used with a ["glob pattern"](https://en.wikipedia.org/wiki/Glob_(programming)). This option has the advantage to work nicely on Windows:

```shell
$ hurl --test --glob 'mydir/foo*.hurl'
```

(note the single quote around the glob value to prevent bash expansion)

We have also an issue to make a ["glob case insentive" option](https://github.com/Orange-OpenSource/hurl/issues/1365) (the current `--glob` is case sensitive). Given these possibilities, unless I miss something, there should be very few use case for a specific directory option.

**Reply by cassepipe** [reply]
Thanks for your reply !

I know I can use the power of the shell but it seems a tad tedious to have to type it all when in a perfect world you could jsut run `hurl  .` or `hurl some_dir`.

Let me change my question:

Are there any reason *not to* implement that feature (apart from time and resources of course) ?

**Comment by jcamiel** [comment]
Ah ah, put in this way effectively there is no strong reason not to add something like this 😅. From your initial question, I was thinking that you were proposing a new option but in fact it's just allowing the use of `hurl .`. In this case, why not!
In your mind, if a user writes `hurl some_dir` do you need to search recursively among subdirectories of `some_dir` (and filtering on `.hurl` extension) ?

**Reply by cassepipe** [reply]
That's a good question, thanks for asking. I don't know, it depends on your time and resources I guess ? You would have to be a heavy `hurl` user to  split your hurl files into sub-directories since the file themselves are already sessions.

I don't have a use for it as of yet but maybe someone will ask for it at some point who knows ?

One thing I see *against* recursive search if I put my implementer's hat on is that now you have to decide what's a good default depth or if you will provide a way to precise a depth or even have to wonder if that will really ever be an issue :sweat_smile:

**Reply by jcamiel** [reply, marked as answer]
I convert the discussion to a new issue as it's definitely a good feature

**Comment by jcamiel** [comment]
Hi @cassepipe it has been a long time but now this feature is completed (will be in Hurl 5.0.0) (see #1446)

**Reply by cassepipe** [reply]
One less script in my repos. Thanks for the follow up !
